pages/api/guilds/[guildId]/configs/[configId].ts

```diff
@@ -0,0 +1,135 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import {
+  DiscordServerConfigRepository,
+  DiscordServerRepository,
+} from "../../../../../db";
+import { validateDashboardToken } from "../../../../../utils/validateDashboardToken";
+import WatchTowerLogger from "../../../../../watchTower";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const { guildId, configId, token } = req.query;
+
+  if (
+    !guildId ||
+    typeof guildId !== "string" ||
+    !configId ||
+    typeof configId !== "string" ||
+    !token ||
+    typeof token !== "string"
+  ) {
+    return res.status(400).json({
+      error: "Missing or invalid guild ID, config ID, or token",
+    });
+  }
+
+  // Validate token
+  const isValidToken = await validateDashboardToken(guildId, token);
+  if (!isValidToken) {
+    return res.status(403).json({
+      error: "Invalid or expired token",
+    });
+  }
+
+  // Check if the guild exists
+  const discordServer = await DiscordServerRepository.findOneBy({
+    id: guildId,
+  });
+
+  if (!discordServer) {
+    return res.status(404).json({
+      error: "Guild not found",
+    });
+  }
+
```

Comment:
For both PUT and GET, you need to check the auth token is valid, else anyone can query and modify the configs